Update Data API reference

mDuo13 opened the pull request:
- Adds new methods that were in the README from the Data API repo
- Fixes some remaining camelCase field names
- other minor changes to sync w/ https://github.com/ripple/rippled-historical-database/pull/172

mDuo13 committed:
Update Data API reference

ryangyoung commented on content/references/data-api.md:
```diff
@@ -24,6 +24,7 @@ The Ripple Data API v2 replaces the Historical Database v1 and the [Charts API](
 [v2.1.0]: https://github.com/ripple/rippled-historical-database/releases/tag/v2.1.0
 [v2.2.0]: https://github.com/ripple/rippled-historical-database/releases/tag/v2.2.0
 [v2.3.0]: https://github.com/ripple/rippled-historical-database/releases/tag/v2.3.0
+[v2.3.2]: https://github.com/ripple/rippled-historical-database/releases/tag/v2.3.2
```
No v2.3.1?

mDuo13 replied:
Correct, [there isn't a v2.3.1](https://github.com/ripple/rippled-historical-database/releases/).

ryangyoung commented on content/references/data-api.md:
```diff
@@ -87,6 +89,7 @@ Validation Network Methods:
 * [Get Validator Validations - `GET /v2/network/validators/{:pubkey}/validations`](#get-validator-validations)
 * [Get Single Validator Reports - `GET /v2/network/validators/{:pubkey}/reports`](#get-single-validator-reports)
 * [Get Daily Validator Reports - `GET /v2/network/validator_reports`](#get-daily-validator-reports)
+* [Get rippled Versions - `GET /v2/network/rippled_versions`](#get-rippled-versions)
```
Should that rippled be `rippled`?

ryangyoung commented on content/references/data-api.md:
```diff
@@ -1944,6 +1947,128 @@ Response:
+| Field  | Value  | Description |
+|--------|--------|-------------|
+| `source` | String | Domain name of the specific external market. |
+| `base_volume` | Number | Exchange volume in terms of the base currency (XRP) |
```
Will this always be XRP? The meaning of the parens here isn't clear to me. Also, I think the description is missing a period (.).

ryangyoung replied:
nitpick: Hmm, looks like all of the descriptions are inconsistent w/r/t to periods. We should probably add to all or remove from all.

mDuo13 replied:
I didn't write this one, but I suspect that this means the base currency is always XRP.

As for the periods, my latest commit should fix that.

ryangyoung commented on content/references/data-api.md:
```diff
@@ -2331,6 +2456,107 @@ Response:
 
 
+
+## Get Fee Stats
```
How are these methods organized? Probably outside the scope of this PR but we might want to readdress this. I was expecting them to be listed in alpha order but got...something else.

mDuo13 replied:
They're partly organized by the URL of the method, but not completely.

I agree we should fix it in a separate task.

ryangyoung commented on content/references/data-api.md:
```diff
@@ -2331,6 +2456,107 @@ Response:
+## Get Fee Stats
+[[Source]<br>](https://github.com/ripple/rippled-historical-database/blob/develop/api/routes/network/getFeeStats.js "Source")
+
+Returns snapshots of the metrics derived from rippled's fee command. _(New in [v2.3.2][])_
```
> rippled's

Should be?

> `rippled`'s 

And should that fee be in code font as well?

ryangyoung reviewed (approved):
Found a few nitpicks but overall looks good!

mDuo13 committed:
Data API style consistencies.

- End description cells with a period
- Use 'The default is' instead of 'Defaults to'
- Auto table column widths
- Put 'rippled' in code font
- Put 'true' and 'false' in code font
- Remove colons from URL templates per RFC6570
- Fix missing column in fee method description
